src/hooks/engine.rs

```diff
@@ -962,6 +990,23 @@ pub struct dma_t {
     pub buffer: *mut c_uchar,
 }
 
+#[repr(C)]
+pub struct cl_entity_s {
+    pub index: c_int,
+    pub padding1: [c_char; 4 + 340 * 3],
+    pub current_position: c_int,
+    pub padding2: [c_char; 1792 + 8 + 52 + 4],
```

Comment:
This kind of long offset is likely different between engine versions, this is what the separate `cls_demos` type pointer is for.